src/stackmap_checker/utils.c

```diff
@@ -3,13 +3,26 @@
 #include <unistd.h>
 #include <sys/mman.h>
 #include <fcntl.h>
+#include <err.h>
+#include <stdlib.h>
 #include "utils.h"
 
-#define TRACE_BIN "trace"
+#define MAX_BUF_SIZE 512
 
-Elf64_Ehdr* get_elf_header(const char *section_name)
+char* get_binary_path()
 {
-    int fd = open(TRACE_BIN, O_RDONLY);
+    char *buff = malloc(MAX_BUF_SIZE);
+    int buff_size = readlink("/proc/self/exe", buff, MAX_BUF_SIZE - 1);
```

Comment:
I'm not too fussed about this, as this code is highly experimental, but not all systems have `/proc`. In a production system you'd have to look at argv[0].

Reply:
I am aware of that, but there is no `argv` in my program, since there is no `main`. This is the only solution I could come up with.